London| May-2025 | Giovanna Bellon | Form control -My T-Shirt

gisbcode opened the pull request:
<!--



[London | MAY-2025 | Giovanna Bellon| Form controls - My T-shirt]

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist
I created a Form for a T-shirt Site

## Questions

gisbcode commented:
Hi, how can I get the completed label?

gisbcode commented:
I have submitted again the Form-control PR, and I made some changes. Let me know if now is under the requirements.
Thanks!

cjyuan commented on Form-Controls/index.html:
```diff
@@ -9,10 +9,47 @@
-      <form>
+      <form method="GET">
+        <!-- Customer Name -->
+  <label for="name">Full Name:</label>
+  <input type="text" id="name" name="name" minlength="2" required />
+  <br /><br />
```
1. Currently a user can enter a name consisting of only space characters (e.g., " "). Can you enforce a stricter validation rule using the `pattern` attribute to disallow any name that contains only space characters?

2. Can you improve the indentation of these few lines of code?

gisbcode replied:
Hi, I have completed the form again. Can you please check it? Also, I have corrected the checked box.
Thanks!

cjyuan replied:
Your current pattern, `pattern="[A-Za-z\s]{2,}"`, still allows a name containing only space characters. (You can improve it later when have time).

gisbcode replied:
I did not understand what name I need to put there

cjyuan reviewed (commented):
Implementation is pretty solid.

1. According to https://validator.w3.org/, there are errors in your code. Can you fix them?

2. To follow best practices, can you format the checked boxes using the proper Markdown syntax in your PR description so that they look something like this?
![image](https://github.com/user-attachments/assets/6f85e6c5-4dd7-49aa-a9fc-e17a9dc2189d)
(With proper Markdown syntax, we can use mouse to check/uncheck the items)

3. A PR branch should include only the changes relevant to the specific task or feature it addresses. This branch was created for the "Form-Controls" exercise; therefore, it should not contain `Wireframe/index.html`, as that file is unrelated to this exercise.
Could you please revert the change made to `Wireframe/index.html`? One way to do this is by replacing the file with the version from your main branch. You can download that version here: https://github.com/gisbcode/Module-Onboarding/tree/main/Wireframe

gisbcode commented:
Hi, all right, I you follow your instructions.
1. I have fixed
3. I did not understand and did not know how to do this number 3 that you said

cjyuan commented:
You replaced the wrong file. You replaced `Form-Controls/index.html` instead of `Wireframe/index.html`.

You can still find your previous `Form-Controls/index.html` in your commit history:
https://github.com/CodeYourFuture/Module-Onboarding/blob/e4e53003783707e76d195daeb306a0d9efdb0736/Form-Controls/index.html
After you restore your `Form-Controls/index.html`, you should still need to fix the error in the HTML code.


To restore `Wireframe/index.html` to its original state, replace it by the version in the `main` branch:
https://github.com/gisbcode/Module-Onboarding/blob/main/Wireframe/index.html

cjyuan commented:
You restored your `Form-Controls/index.html`. Good job! 

1. According to https://validator.w3.org/, there are errors in your code. Can you fix them? (What errors do you see when you check your code on W3C Markup Validation Service?)

2. To restore `Wireframe/index.html` to its original state, replace it by the version in the main branch:
https://github.com/gisbcode/Module-Onboarding/blob/main/Wireframe/index.html
    - Download `Wireframe/index.html` from the above link
    - Use the downloaded file to replace the same file in this branch.
    - Commit the change.

cjyuan commented:
1. When I validated your `Form-Controls.html` at https://validator.w3.org/, it shows me these:
![image](https://github.com/user-attachments/assets/612cce2c-c05e-42f1-840a-e74b7586315c)
You can use the online validator to check your code to ensure there is not errors and warnings in your code.

2. I was not checking your Wireframe implementation. I meant to say, you should NOT have modified `index.html` in the "Wireframe" folder. GitHub currently shows two modified files in the "Files changed" tab:
https://github.com/CodeYourFuture/Module-Onboarding/pull/645/files
I expected only files in the "Form-Controls" folder are modified, but I saw a file in the "Wireframe" folder is also modified. That's why I asked you to restore the file in the "Wireframe" folder so that it won't be identified as a modified file by GitHub.

![image](https://github.com/user-attachments/assets/1497fc86-9c28-4595-9089-e70d1a9155a4)

cjyuan commented:
I don't see any new commits made on this branch since my last review. Have you push your changes to Github?

gisbcode commented:
I added the Headings in line 20, 35 and 46. Also, I put the spaces that it was required in the error board that you showed me.

cjyuan commented:
I cannot see what you change on your computer if you don't commit your changes and push the commits to GitHub.

You can see all the commits you pushed to this branch here:
https://github.com/CodeYourFuture/Module-Onboarding/pull/645/commits

The most recent commits was made 3 hours ago.

This is what you page looks like currently: https://deploy-preview-645--cyf-onboarding-module.netlify.app/form-controls/

cjyuan commented:
It seems you have made some changes in the "Form-Controls" folder in your `Feature/Wireframe` branch. We need to pay attention to "what is the current branch" when we make changes.

![image](https://github.com/user-attachments/assets/03f9b708-4f18-4210-8311-3eea0e8e1062)

gisbcode commented:
Sorry for my mistake, I fixed it now in the right place. I added the headings and put the spaces where it was required.

cjyuan commented:
You should also add the "Needs review" label to indicate that the changes you made is ready to be reviewed. (The reviewer may miss the notification, and without the label, CYF or other volunteers may not be aware of your PR).

gisbcode commented:
I have checked before when you said first time, and the page did not work for me. 

I have already reverted and fixed the Wireframe index, I'm not understanding why you are asking this.

gisbcode commented:
I just fixed some other errors. Let me know how it is now.

cjyuan commented:
The errors and warnings are fixed. (The spec says, "No CSS", but you don't have to undo it this time) 

Let's try to resolve the modified `Wireframe/index.html` issue so that you can complete this PR.

If you have successfully fixed `Wireframe/index.html`, then the file will not show up in the "Files changed" tab in this PR:
https://github.com/CodeYourFuture/Module-Onboarding/pull/645/files
![image](https://github.com/user-attachments/assets/c92101d0-aee6-41a5-9ea8-c86f13704326)

cjyuan commented:
I see the problem now: You modified `index.html` in your `main` branch 3 weeks ago.

Can you replace `Wireframe/index.html` using the version in CYF's `main` (instead of your own `main` branch).
https://github.com/CodeYourFuture/Module-Onboarding/blob/main/Wireframe/index.html

gisbcode commented:
Thank you for that,I still could not see where the error is in the branch. I'm really tired, so I can fix this other time, but I need to submit the exercise link with the complete label today.

cjyuan commented:
This PR was already marked "Complete". 
Course Platform is downed today. You probably cannot submit the step tonight but you can prepare the issue and submit the issue link when the server is up.

Here is the instruction for preparing the issue to be submitted in step 1: https://github.com/CodeYourFuture/Module-Onboarding/issues/570

gisbcode commented:
I noticed that, I hope tomorrow works once is the deadline day. Thanks for sharing the link! Thanks for the help and patience!
